src/Orleans/Utils/DictionaryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans
+{
+    static class DictionaryExtensions
+    {
+        public static TResult Find<TKey, TResult>(this IDictionary<TKey, TResult> dictionary, TKey key) where TResult : class
```

Comment:
These are rather self-explanatory, but should they be commented? So as to set the tone for the code. :) ``IReadOnlyDictionary``?